### ISSUE TITLE
Manipulating the terminal causes tests to be stuck

### ISSUE THREAD

**Issue by sigmaSd** [nextest-rs/nextest#945]
code:

`cargo add crossterm`
```rs
#[test]
fn t() {
    // both of these freeze cargo next-test run
    // crossterm::terminal::enable_raw_mode().unwrap();
    // crossterm::cursor::position().unwrap();
}
```

**Comment by sunshowers** [comment]
Thanks. This is a good flag -- what's happened is that this causes the child to raise a SIGTTOU signal on itself, and the default behavior of that signal is to put the process in a stopped state (similar to ctrl-Z).

* At the very least, nextest definitely should ensure that if a test gets into a stopped state, termination continues to work.
* There was a suggestion at some point to use a [pty](https://man7.org/linux/man-pages/man7/pty.7.html) for each test. We didn't do it for complexity reasons but maybe we should implement it as an option, and let tests opt into it.

**Comment by sunshowers** [comment]
> I wish there were a way to change the signal handler for SIGTTOU/SIGTTIN to terminate the process, but I don't believe there is.

Probably too intrusive for a test runner, to be honest. I think the out-of-process mechanics used by nextest (termination, timeouts) should generally be the extent of what we do.

The pty would definitely be interesting as well.

**Comment by ClementTsang** [comment]
Recently tried using the [portable_pty crate](https://docs.rs/portable-pty/latest/portable_pty/) (commit [here](https://github.com/ClementTsang/bottom/pull/1383/commits/860e5c2a0af5e594368282c2bc8aaee3e711d379)) to test a TUI application; nextest seems to work with it.

**Comment by sunshowers** [comment]
Yes, if we allow using a pty for each test, then I was eyeing the portable_pty crate to make that happen.
